src/NzbDrone.Common/Composition/AssemblyLoader.cs

```diff
@@ -28,8 +28,9 @@ public static IEnumerable<Assembly> Load(IEnumerable<string> assemblyNames)
 
             var startupPath = AppDomain.CurrentDomain.BaseDirectory;
 
-            return toLoad.Select(x =>
-                AssemblyLoadContext.Default.LoadFromAssemblyPath(Path.Combine(startupPath, $"{x}.dll")));
+            return toLoad
+                .Select(x => AssemblyLoadContext.Default.LoadFromAssemblyPath(Path.Combine(startupPath, $"{x}.dll")))
+                .ToArray();
```

Comment:
Why `ToArray()` instead of `ToList()` since we're returning an `IList` now?

Reply:
@markus101, both of them inherit IList, so we can return any of them. I choose arrays whenever there are no plans to add anything. Let's use `ToList` if you prefer (to be honest, it will be more consistent, because the project mainly uses lists).

Reply:
@Qstick, In terms of performance ToArray and ToList are quite similar. If the source is IEnumberable without the ability to determine the size:
```
|        Method |           Mean |        Error |       StdDev |     Gen0 |     Gen1 |     Gen2 | Allocated |
|-------------- |---------------:|-------------:|-------------:|---------:|---------:|---------:|----------:|
|  SmallToArray |       234.6 ns |      0.38 ns |      0.35 ns |   0.0038 |        - |        - |     328 B |
|   SmallToList |       219.8 ns |      0.81 ns |      0.76 ns |   0.0038 |        - |        - |     328 B |
| MediumToArray |     1,520.3 ns |      5.56 ns |      5.20 ns |   0.0134 |        - |        - |    1264 B |
|  MediumToList |     1,452.4 ns |      6.25 ns |      5.85 ns |   0.0153 |        - |        - |    1296 B |
|  LargeToArray |    13,632.9 ns |     90.09 ns |     79.87 ns |   0.0916 |        - |        - |    8608 B |
|   LargeToList |    12,689.3 ns |     85.01 ns |     79.52 ns |   0.0916 |        - |        - |    8536 B |
|   HugeToArray | 1,427,933.8 ns | 13,588.98 ns | 12,711.14 ns | 248.0469 | 248.0469 | 248.0469 |  925289 B |
|    HugeToList | 1,383,763.2 ns |  3,167.48 ns |  2,962.86 ns | 285.1563 | 285.1563 | 285.1563 | 1049281 B |
```

And when the source is an collection with known size:
```
|        Method |         Mean |      Error |     StdDev |     Gen0 |     Gen1 |     Gen2 | Allocated |
|-------------- |-------------:|-----------:|-----------:|---------:|---------:|---------:|----------:|
|  SmallToArray |     21.43 ns |   0.037 ns |   0.031 ns |   0.0007 |        - |        - |      64 B |
|   SmallToList |     26.11 ns |   0.087 ns |   0.077 ns |   0.0011 |        - |        - |      96 B |
| MediumToArray |     55.41 ns |   0.337 ns |   0.299 ns |   0.0051 |        - |        - |     424 B |
|  MediumToList |     58.97 ns |   0.414 ns |   0.367 ns |   0.0054 |        - |        - |     456 B |
|  LargeToArray |    385.27 ns |   3.799 ns |   3.554 ns |   0.0477 |        - |        - |    4024 B |
|   LargeToList |    380.38 ns |   1.231 ns |   1.152 ns |   0.0482 |        - |        - |    4056 B |
|   HugeToArray | 81,527.71 ns | 170.190 ns | 159.196 ns | 124.8779 | 124.8779 | 124.8779 |  400108 B |
|    HugeToList | 81,751.91 ns | 761.364 ns | 674.929 ns | 124.8779 | 124.8779 | 124.8779 |  400140 B |

```

P.S. Small means 10 elements, medium - 100, large - 1000, huge - 100000